src/codegen/doc/mod.rs

```diff
@@ -44,46 +44,115 @@ pub fn generate(env: &Env) {
         |w| generate_doc(w, &env));
 }
 
+pub struct TypeReferences {
+    refs: Vec<TypeStruct>,
+}
+
+impl TypeReferences {
+    pub fn new() -> TypeReferences {
+        TypeReferences { refs: vec!() }
+    }
+
+    pub fn add(&mut self, ty: TypeStruct) {
+        if let Some(_) = self.get_type(&ty.name) {}
+        else {
+            self.refs.push(ty);
+        }
+    }
+
+    pub fn get_type(&self, type_name: &str) -> Option<TypeStruct> {
+        for ty in self.refs.iter() {
+            if &ty.name == type_name {
+                return Some(ty.clone())
+            }
+        }
+        None
+    }
+}
```

Comment:
That's what I did at first. Not so sure what the better solution is. A `HashMap` would duplicate data.